runtime.ts

```diff
@@ -162,7 +162,14 @@ export function resolveModule(
   if (m != null) {
     return m;
   } else {
-    return new FileModule(filename, sourceCode, outputCode);
+    // TODO: keep imports/exports, functions & classes out out
+    // TODO: only wrap when await is used
+    // TODO; apply transform at compiler level
+    const wrappedSource = /await/g.test(sourceCode)
```

Comment:
deno uses this code to load it's own modules too, (add a `console.log(filename)` to see what I mean : )); so I think you should check if it's an internal asset or not...
btw, why not do this in Go? see `HandleCodeFetch()` function
```
if isRemote(moduleName) {
```
you add this wrapper to users code in last else statement of the above if-condition

Reply:
Thank you! btw you can check out this [gist](https://gist.github.com/qti3e/fc3a2271b2577126607349a405753abb)